smac/smbo/pSMAC.py

```diff
@@ -29,27 +38,30 @@ def read(run_history, output_directory, configuration_space, logger):
     numruns_in_runhistory = len(run_history.data)
     initial_numruns_in_runhistory = numruns_in_runhistory
 
-    files_in_output_directory = os.listdir(output_directory)
-    for file_in_output_directory in files_in_output_directory:
-        match = re.match(RUNHISTORY_RE, file_in_output_directory)
-        if match:
-            runhistory_file = os.path.join(output_directory,
-                                           file_in_output_directory)
-            run_history.update_from_json(runhistory_file,
-                                         configuration_space)
-
-            new_numruns_in_runhistory = len(run_history.data)
-            difference = new_numruns_in_runhistory - numruns_in_runhistory
-            logger.debug('Shared model mode: Loaded %d new runs from %s' %
-                         (difference, file_in_output_directory))
-            numruns_in_runhistory = new_numruns_in_runhistory
+    if isinstance(output_dirs, str):
+        output_dirs = glob.glob(output_dirs)
+
+    for output_directory in output_dirs:
+        for file_in_output_directory in os.listdir(output_directory):
```

Comment:
Why no glob.glob here?

Maybe first go over all entries in output_dirs to create one list with all directories found with glob